releng/run.sh

```diff
@@ -17,6 +17,17 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+# Auto-detect architecture
+if [[ "${GOARCH:-}" == "" ]]; then
+  HOST_ARCH=$(go env GOARCH)
+fi
+
+# Always build for Linux, but use native architecture
```

Comment:
Can we add a note why do we always build for Linux? Is this script used exclusively in a Linux container?